keps/sig-auth/4412-projected-service-account-tokens-for-kubelet-image-credential-providers/README.md

```diff
@@ -0,0 +1,1079 @@
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+As Kubernetes has matured, we have taken strides towards limiting the need for long
+lived credentials that are stored in the Kubernetes API.  The canonical example of this
+change is the move of Kubernetes Service Account (KSA) tokens from ones that were
+long lived, persisted in the Kubernetes API, and never rotated to tokens that are ephemeral,
+short lived, and automatically rotated.  By hardening these tokens and stabilizing
+their schema to match the semantics of OIDC ID tokens, we have enabled external consumers
+to validate these tokens for use cases beyond Kubernetes.  For example, it is possible to
+use a KSA token that is bound to a specific pod to fetch secrets from an external cloud
+provider vault without the need for any long lived secrets in the cluster.
+
+This KEP aims to make it possible for a similar 'secret-less' flow to be created for
+image pulls, which happen before the pod is running.  Today, admins are limited to image pull
+secrets that are stored directly in the Kubernetes API and thus are long lived and hard to rotate,
+or secrets that are managed at the Kubelet level via a Kubelet credential provider (meaning
+that any pod running on that node can access those images).  A pod should instead be able
+to use its own identity to pull images, i.e. we should enable image pull authorization to be
+tied to a particular workload while avoiding the need for long lived, persisted secrets.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+By moving to image pulls that are based on KSA tokens, we seek to:
+
+- Reduce secret management overhead for developers and admins
+- Reduce security risks associated with long lived secrets
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Allow workloads to pull images based on their own runtime identity without long lived / persisted secrets
+- Avoid needing a kubelet/node based identity to pull images
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Passthrough mode where KSA tokens are sent directly to the CRI
```

Comment:
maybe move from "non-goals" to "possible future work", maybe with a sketch of what it would look like ... not sure we care about making it part of initial graduation criteria for this, but ensuring we don't block expanding in this direction would be ideal

Reply:
done!